runtime/drivers/athena/sql_store.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		return err
+	}
+
+	tm := time.NewTimer(5 * time.Minute)
+	defer tm.Stop()
+	for {
+		select {
+		case <-tm.C:
+			return fmt.Errorf("Athena ingestion timed out")
+		default:
+			status, err := client.GetQueryExecution(ctx, &athena.GetQueryExecutionInput{
+				QueryExecutionId: queryExecutionOutput.QueryExecutionId,
+			})
+			if err != nil {
+				return err
+			}
+
+			switch status.QueryExecution.Status.State {
+			case types2.QueryExecutionStateSucceeded, types2.QueryExecutionStateCancelled:
+				return nil
+			case types2.QueryExecutionStateFailed:
+				return fmt.Errorf("Athena query execution failed %s", *status.QueryExecution.Status.AthenaError.ErrorMessage)
+			}
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func parseSourceProperties(props map[string]any) (*sourceProperties, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return conf, nil
+}
+
+func resolveOutputLocation(ctx context.Context, client *athena.Client, conf *sourceProperties) (string, error) {
+	if conf.OutputLocation != "" {
+		return conf.OutputLocation, nil
+	}
+
+	workgroup := conf.Workgroup
+	// fallback to "primary" (default) workgroup if no workgroup is specified
+	if workgroup == "" {
+		workgroup = "primary"
+	}
+
+	wo, err := client.GetWorkGroup(ctx, &athena.GetWorkGroupInput{
+		WorkGroup: aws.String(workgroup),
+	})
+	if err != nil {
+		return "", err
+	}
+
+	resultConfiguration := wo.WorkGroup.Configuration.ResultConfiguration
+	if resultConfiguration != nil && resultConfiguration.OutputLocation != nil && *resultConfiguration.OutputLocation != "" {
+		return *resultConfiguration.OutputLocation, nil
+	}
+
+	return "", fmt.Errorf("either output_location or workgroup with an output location must be set")
+}
+
+func openBucket(ctx context.Context, cfg aws.Config, bucket string) (*blob.Bucket, error) {
+	s3client := s3.NewFromConfig(cfg)
+	return s3blob.OpenBucketV2(ctx, s3client, bucket, nil)
+}
+
+func deleteObjectsInPrefix(ctx context.Context, cfg aws.Config, bucketName, prefix string) error {
+	s3client := s3.NewFromConfig(cfg)
+
+	deleteBatch := func(objects []types.ObjectIdentifier) error {
+		_, err := s3client.DeleteObjects(ctx, &s3.DeleteObjectsInput{
+			Bucket: &bucketName,
+			Delete: &types.Delete{
+				Objects: objects,
+			},
+		})
+		return err
+	}
+
+	var continuationToken *string
+	for {
+		out, err := s3client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+			Bucket:            &bucketName,
+			Prefix:            &prefix,
+			ContinuationToken: continuationToken,
+		})
+		if err != nil {
+			return err
+		}
+
+		ids := make([]types.ObjectIdentifier, 0, len(out.Contents))
+		for _, o := range out.Contents {
+			ids = append(ids, types.ObjectIdentifier{
+				Key: o.Key,
+			})
+		}
+
+		if len(ids) > 0 {
+			if err := deleteBatch(ids); err != nil {
+				return err
+			}
+		}
+
+		if out.IsTruncated {
+			continuationToken = out.NextContinuationToken
```

Comment:
Maybe also check `out.NextContinuationToken != nil` just to be extra cautious that we don't loop forever?